src/XIVLauncher.Common.Unix/Compatibility/DxvkSettings.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace XIVLauncher.Common.Unix.Compatibility;
+
+public class DxvkSettings
+{
+    public string DownloadURL { get; private set; }
+
+    public string FolderName { get; private set; }
+
+    public Dictionary<string, string> DxvkVars { get; private set; }
+
+    public Dxvk.DxvkHudType DxvkHud { get; private set; }
+
+    public Dxvk.DxvkVersion DxvkVersion { get; private set; }
+
+    private const string ALLOWED_CHARS = "^[0-9a-zA-Z,=.]+$";
+    private const string ALLOWED_WORDS = "^(?:devinfo|fps|frametimes|submissions|drawcalls|pipelines|descriptors|memory|gpuload|version|api|cs|compiler|samplers|scale=(?:[0-9])*(?:.(?:[0-9])+)?)$";
+
+    public DxvkSettings(Dxvk.DxvkHudType hud = Dxvk.DxvkHudType.None, string dxvkHudCustom="",
+        string mangoHudPath="", bool? async = true, int? frameRate = 0,
+        Dxvk.DxvkVersion version = Dxvk.DxvkVersion.v1_10_1, string? corePath = null)
+    {
+        this.DxvkHud = hud;
+        string home = Environment.GetEnvironmentVariable("HOME");
+        corePath ??= Path.Combine(home,".xlcore");
+        string dxvkConfigPath = Path.Combine(corePath,".dxvk");
```

Comment:
Maybe it is cleaner to pass a DirectoryInfo into the cctor here as well, just in case the base xlcore storage path will change

Reply:
That's just the default. You can pass in the correct path to the constructor; it just defaults to null and sets the default to ~/.xlcore in that case. XL Core's Program.cs creates a DxvkSettings object, and I've passed it in there. But I could change it to pass in a DirectoryInfo. It's probably more "correct". So now the storage.Root object can be passed in from XL Core, instead of the string storage.Root.FullName.